test/unit/ids.js

```diff
@@ -16,26 +17,31 @@ const mockDb = (idFilterLogicFn) => {
 };
 
 exports.tearDown = callback => {
-    testUtils.restore([
-        Math.random
-    ]);
+  testUtils.restore([
+    Math.random
+  ]);
 
-    callback();
+  callback();
 };
 
-exports['generates an id of the given length'] = function(test) {
-    [1,2,3,4,5,6,7,8,9,10].forEach(function(l) {
-        test.equal(ids._generate(l).length, l);
-    });
+exports['generates an id of the given length'] = test => {
+  [1, 2, 3, 4, 5, 6, 7, 8, 9, 10].forEach(l =>
+    test.equal(ids._generate(l).length, l));
 
-    test.done();
+  test.done();
 };
 
-exports['ids can start with 0, will be correct length'] = function(test) {
-    sinon.stub(Math, 'random').returns(0.00001);
+exports['ids can start with 0, will be correct length'] = test => {
+  sinon.stub(Math, 'random').returns(0.00001);
 
-    test.equal(ids._generate(5), '00000');
-    test.done();
+  test.equal(ids._generate(5), '00000');
+  test.done();
+};
+
+exports['ids are "always" the length they should be'] = test => {
+  test.ok(
+    jsc.checkForall(jsc.integer(5, 13), i => ids._generate(i).length === i));
```

Comment:
I agree it's hard to tell from one line!

In terms of whether we need a library, I'll say two things:
 - If you're asking if we should be doing property based testing at all, then the answer is I don't know. I think it could be handy in some very specific cases, specifically around automated bounds testing of more "algorithm-y" things
 - If you're asking if we need a library in general because dependencies are bad, then IDK, I mean we don't really need mocha / nodeunit / jasmine either honestly, they just execute JS and catch exceptions. 

It may be worthwhile reading a bit about QuickCheck and friends, and they are bit weird and do a bit more than may be obvious.

FWIW I may be adding some around dates in the near future that will probably look more impressive, and will probably be a more useful evaluation of its worth.